metals/src/main/scala/scala/meta/internal/worksheets/WorksheetProvider.scala

```diff
@@ -120,11 +119,12 @@ class WorksheetProvider(
       }
     }
     val onError: PartialFunction[Throwable, Option[EvaluatedWorksheet]] = {
-      case NonFatal(e) =>
-        scribe.error(s"worksheet: $path", e)
-        None
       case InterruptException() =>
         None
+      case e: Throwable =>
+        // NOTE(olafur): we catch all exceptions because of https://github.com/scalameta/metals/issues/1456
```

Comment:
catching all exceptions is handled in mdoc and they are converted to NonFatal errors which are already handled, this case is not related to mentioned issue, right?

Reply:
It's correct that they should be caught by mdoc now but I added this case nevertheless to be sure. If there's something fatally broken about the JVM I suspect it will show up in the logs regardless.